File: lib/typescript/botbuilder-skills/src/auth/microsoftAppCredentialsEx.ts
/**
 * Copyright(c) Microsoft Corporation.All rights reserved.
 * Licensed under the MIT License.
 */

import { MicrosoftAppCredentials } from 'botframework-connector';

export class MicrosoftAppCredentialsEx extends MicrosoftAppCredentials {
    constructor(appId: string, password: string, oauthScope?: string) {
        super(appId, password);
        if (oauthScope) {
            this.oAuthScope = oauthScope;
        }
        
        this.oAuthEndpoint = 'https://login.microsoftonline.com/microsoft.com';
    }
}

File: sdk/csharp/libraries/microsoft.bot.builder.solutions/Skills/Auth/MicrosoftAppCredentialsEx.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

using System;
using Microsoft.Bot.Connector.Authentication;

namespace Microsoft.Bot.Builder.Solutions.Skills.Auth
{
    [Obsolete("This type is being deprecated. To continue using Skill capability please refer to https://aka.ms/botframework-solutions/releases/0_8", false)]
    public class MicrosoftAppCredentialsEx : MicrosoftAppCredentials, IServiceClientCredentials
    {
        private readonly string _oauthScope;

        public MicrosoftAppCredentialsEx(string appId, string password, string oauthScope)
            : base(appId, password)
        {
            _oauthScope = oauthScope;
        }

        public override string OAuthScope => _oauthScope;

        public override string OAuthEndpoint => "https://login.microsoftonline.com/microsoft.com";
    }
}